From d9780a4ed870990f30117f1e3be990e34af072de Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Sun, 3 May 2026 23:01:20 +0800
Subject: [PATCH 417/441] AOSCOS: drm: amdgpu: disable Panel Self Refresh v1
 and v2 (PSR-SU), and Panel Replay

There has been a lot of report about freezing displays on laptops,
especially on those based on RDNA 2/3/4 architectures. These features are
probably more trouble than they are worth, so disable them by default.

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c
index 26b48c010097..53d196ef53a4 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c
@@ -863,7 +863,8 @@ module_param_named(dcfeaturemask, amdgpu_dc_feature_mask, uint, 0444);
  * DOC: dcdebugmask (uint)
  * Display debug options. See enum DC_DEBUG_MASK in drivers/gpu/drm/amd/include/amd_shared.h.
  */
-MODULE_PARM_DESC(dcdebugmask, "all debug options disabled (default))");
+uint dcdebugmask = 0x610;
+MODULE_PARM_DESC(dcdebugmask, "Disable debug options (default: disable Panel Self Refresh v1 and PSR-SU, or v2, as well as Panel Replay)");
 module_param_named(dcdebugmask, amdgpu_dc_debug_mask, uint, 0444);
 
 MODULE_PARM_DESC(visualconfirm, "Visual confirm (0 = off (default), 1 = MPO, 5 = PSR)");
-- 
2.55.0

